.github/workflows/ubuntu.yml

```diff
@@ -205,29 +211,16 @@ jobs:
           cd ../
           rm -r compiled-planner-${{ matrix.ubuntu-version }}
 
-      - name: setup-dependencies
-        run: |
-          pip3 install pytest tox
-          sudo apt-get update
-          sudo apt-get -y install clang-tidy-8 valgrind zlib1g-dev libgmp3-dev
-          sudo apt-get -y install ${{ matrix.compiler-version }}
-
       - name: test
         env:
+          CC: ${{ matrix.compiler-version.cc }}
+          CXX: ${{ matrix.compiler-version.cxx }}
           DOWNWARD_CPLEX_ROOT: /home/runner/work/downward/lib/ibm/ILOG/CPLEX_Studio129/cplex
           DOWNWARD_SOPLEX_ROOT: /home/runner/work/downward/lib/soplex-3.1.1
           DOWNWARD_COIN_ROOT: /home/runner/work/downward/lib/coin
         run: |
           export PATH="$(pwd):$PATH"  # Add uncrustify and VAL to PATH.
           cd misc/
```

Comment:
One more thing: I think all tests that don't require compilation (style, clang-tidy, translator) should be run before the compilation. The style and clang-tidy tests are the ones that most frequently fail and it makes no sense to spend all the cycles for useless compilations. Quicker feedback will also speed up our development process.

Reply:
That makes sense. However, I would like to figure out first what exactly we want to test under which setting. Do we need the tests you mentioned here only once, with any python version? Does it matter at all?

Reply:
My 5cents:
- Everything that we run to verify that a functionality works (i.e. translator tests) should be run with all relevant python versions (see comments above about which python we need to test actually)
- Everything that ensures code quality is sufficient to run on a single python version (whether the script fails on some python version is not important, important is to know if the code quality is right). Ensuring that those scripts work on any python version is nice for the developers, because we can use it on our machines (although, once it breaks with a new python version, we would just push a fix)

Reply:
Ideally, the style and clang-tidy tests would only run once per push (not once per OS or once per matrix config) and all other jobs would only execute if they're successful. That would require, however, that all jobs are in the same workflow (which makes for a long file, but I think that's negligible).

Reply:
Agreed on the "only run once per push". But is there really a good motivation for not trying the other tests if style or clang-tidy fail?

Reply:
Usually, the failing tests will need to be run again anyway and I think it makes sense to save those CPU cycles.

Reply:
If I pushed something in order to get it tested, then came back a few hours later to see that the tests didn't run because I inserted an extra newline, I'd probably be annoyed. But if the people that work on the code more than me prefer this, I have no objections.